builtin_array.go

```diff
@@ -794,6 +795,7 @@ func (r *Runtime) arrayproto_map(call FunctionCall) Value {
 					values[k] = callbackFn(fc)
 				}
 			}
+			r.vm.tmpValues = nil
```

Comment:
`nil` out the tmpValues array because the values array will now be tracked when this function returns and the returned array object is pushed onto the stack.